src/stdafx.h

```diff
@@ -428,9 +429,12 @@ void NORETURN CDECL error(const char *str, ...) WARN_FORMAT(1, 2);
 #	define assert(expression) if (!(expression)) error("Assertion failed at line %i of %s: %s", __LINE__, __FILE__, #expression);
 #endif
 
-/* Asserts are enabled if NDEBUG isn't defined or WITH_ASSERT is defined. */
-#if !defined(NDEBUG) || defined(WITH_ASSERT)
+/* Asserts are enabled if NDEBUG isn't defined, or if we are using MSVC and WITH_ASSERT is defined. */
+#if !defined(NDEBUG) || (defined(_MSC_VER) && defined(WITH_ASSERT))
```

Comment:
Why does this become MSVC only? On Linux and MacOS `WITH_ASSERT` is also set.

Reply:
Not sure. @JGRennison ?

Reply:
See the next comment ;) I figured it out for you :) It is not @JGRennison I think .. more like old code not being backported properly :P :P :D <3